torchao/prototype/quantization/safetensors_support.py

```diff
@@ -0,0 +1,184 @@
+        raise ValueError("No tensors found")
+
+    tensor_names = json.loads(metadata["tensor_names"])
+    result = {}
+
+    for tensor_name in tensor_names:
+        tensor_metadata = {}
+        for key, value in metadata.items():
+            if key.startswith(f"{tensor_name}:"):
+                # Remove the prefix
+                tensor_metadata[key[len(tensor_name) + 1 :]] = value
+
+        tensor_tensors = {}
+        for key, value in loaded_tensors.items():
+            if key.startswith(f"{tensor_name}:"):
+                # Remove the prefix
+                tensor_tensors[key[len(tensor_name) + 1 :]] = value
+
+        tensor_type = tensor_metadata.get("tensor_type")
+
+        if tensor_type == "Float8Tensor":
+            dtype = ALLOWED_QUANT_DTYPES.get(tensor_metadata.get("dtype"))
+            granularity = ALLOWED_GRANUALARITY.get(tensor_metadata.get("granularity"))
+            kernel_preference = json.loads(tensor_metadata.get("kernel_preference"))
+            hp_value_lb = tensor_metadata.get("hp_value_lb")
+            hp_value_ub = tensor_metadata.get("hp_value_ub")
+            mm_config = Float8MMConfig(*json.loads(tensor_metadata.get("mm_config")))
+
+            from torchao.quantization import Float8Tensor
+
+            result[tensor_name] = Float8Tensor(
+                qdata=tensor_tensors["qdata"].to(tensor_metadata["qdata_device"]),
+                scale=tensor_tensors["scale"].to(tensor_metadata["scale_device"]),
+                block_size=json.loads(tensor_metadata.get("block_size")),
+                mm_config=mm_config,
+                hp_value_lb=hp_value_lb,
+                hp_value_ub=hp_value_ub,
+                act_quant_kwargs=QuantizeTensorToFloat8Kwargs(
+                    float8_dtype=dtype,
+                    granularity=granularity if granularity else PerRow(),
+                    mm_config=mm_config,
+                    hp_value_lb=hp_value_lb,
+                    hp_value_ub=hp_value_ub,
+                    kernel_preference=kernel_preference,
+                ),
+                kernel_preference=kernel_preference,
+                dtype=dtype,
+            )
+        elif tensor_type == "Tensor":
+            data = tensor_tensors["data"]
+            data = data.to(tensor_metadata["device"])
+            result[tensor_name] = data
+
+    return result
+
+
+def create_metadata_for_tensor_subclass(
+    tensor: torch.Tensor,
+) -> Tuple[Dict[str, str], Dict[str, torch.Tensor]]:
+    """
+    Create metadata for tensor subclasses from torchao.
+
+    Args:
+        tensor: A tensor subclass (e.g., Float8Tensor)
+
+    Returns:
+        Tuple of (metadata, tensors_dict) where:
+        - metadata: Dictionary with metadata needed to reconstruct the tensor
+        - tensors_dict: Dictionary with tensors to save
+    """
+    metadata = {}
+    tensors_dict = {}
+
+    if tensor.__class__.__name__ == "Float8Tensor":
+        metadata["tensor_type"] = "Float8Tensor"
```

Comment:
I feel we can do a full path here, might be more robust (torchao.quantization.Float8Tensor), also can we get this from tensor itself instead of hardcode? (not sure what is `tensor.__class__` but something like that might give the full path)